dns/ddclient/src/opnsense/scripts/ddclient/lib/address.py

```diff
@@ -56,6 +57,8 @@ def extract_address(txt):
     for regexp in [r'[^a-fA-F0-9\:]', r'[^F0-9\.]']:
         for line in re.sub(regexp, ' ', txt).split():
             if line.count('.') == 3 or line.count(':') >= 2:
+                if line == '1.1.1.1':
```

Comment:
this doesn't look very portable and well documented...

Reply:
Issue to be tackled is that there is an IP address in the response body, before the actual client IP address.

Options are either to use: [https://one.one.one.one/cdn-cgi/trace](https://one.one.one.one/cdn-cgi/trace) which works with the existing scan logic out of the box but requires an IP lookup.

Or ignore "1.1.1.1" as a client IP address as per above. Technically 1.1.1.1 is fixed to this service by default and hence would never be a valid client IP for any responses.

Based on your feedback, I guess the DNS lookup penalty could be incurred. Updated changes submitted.